File: PRACTICA1/JAVA/deepspace/src/deepspace/SuppliesPackage.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author celia
 */
class SuppliesPackage {
    private float ammoPower;
    private float fuelUnits;
    private float shieldPower;
    
    SuppliesPackage(float ammo, float fuel, float shield){
        ammoPower = ammo;
        fuelUnits = fuel;
        shieldPower = shield;
    }
    
    SuppliesPackage(SuppliesPackage otro){
        ammoPower = otro.ammoPower;
        fuelUnits = otro.fuelUnits;
        shieldPower = otro.shieldPower;
    }
    
    public float getAmmoPower(){
        return ammoPower;
    }
    public float getFuelUnits(){
        return fuelUnits;
    }
    public float getShieldPower(){
        return shieldPower;
    }
    
    public String toString(){
        String mensaje = "El SuppliesPackage tiene de municiones de energía: "+ammoPower+", de combustible: "+fuelUnits+" y de poder de escudo: "+shieldPower+".";
        return mensaje;
    }
}

File: PRACTICA1/RUBY/Deepspace/SuppliesPackage.rb
#encoding:utf-8

module Deepspace


# @author Lucía Salamanca López, Celia Arias Martínez
class SuppliesPackage
  def initialize(ammoPower,fuelUnits,shieldPower)
    @ammoPower = ammoPower
    @fuelUnits = fuelUnits
    @shieldPower = shieldPower
  end

  def ammoPower
    @ammoPower
  end

  def fuelUnits
    @fuelUnits
  end

  def shieldPower
    @shieldPower
  end

  def self.newCopy(orig)
    new(orig.ammoPower, orig.fuelUnits, orig.shieldPower)
  end

  def to_s
    "El SuppliesPackage tiene de municiones de energía: "+@ammoPower.to_s+", de combustible: "+@fuelUnits.to_s+" y de poder de escudo: "+@shieldPower.to_s+"."
  end


end

end
